editor/src/clj/editor/cljfx_form_view.clj

```diff
@@ -423,35 +424,56 @@
-            added-count (count resources)
-            indices (vec (range value-count
-                                (+ value-count added-count)))]
-        [[:dispatch (assoc on-added :fx/event resources)]
-         [:set-ui-state (assoc-in ui-state (conj state-path :selected-indices) indices)]]))))
+(defn- absolute-or-maybe-proj-path
+  ^String [^File file workspace in-project]
+  (if in-project
+    (workspace/as-proj-path workspace file)
+    (.getAbsolutePath file)))
+
+(defn- add-list-elements [added-list-elements {:keys [on-added state-path ui-state value]}]
+  (let [old-element-count (count value)
+        new-element-count (+ old-element-count (count added-list-elements))
+        added-indices (vec (range old-element-count new-element-count))]
+    [[:dispatch (assoc on-added :fx/event added-list-elements)]
+     [:set-ui-state (assoc-in ui-state (conj state-path :selected-indices) added-indices)]]))
+
+(defmethod handle-event :add-list-element [{:keys [element fx/event project workspace] :as map-event}]
+  (case (:type element)
+    :directory
+    (when-some [selected-directory (dialogs/make-directory-dialog
+                                     (or (:title element) "Select Directory")
+                                     (workspace/project-path workspace)
+                                     (fxui/event->window event))]
+      (if-some [valid-directory-path (absolute-or-maybe-proj-path selected-directory workspace (:in-project element))]
+        (add-list-elements [valid-directory-path] map-event)
+        {:show-dialog :directory-not-in-project}))
+
+    :file
+    (when-some [selected-file (dialogs/make-file-dialog
+                                (or (:title element) "Select File")
+                                (:filter element)
+                                nil
+                                (fxui/event->window event))]
+      (if-some [valid-file-path (absolute-or-maybe-proj-path selected-file workspace (:in-project element))]
+        (add-list-elements [valid-file-path] map-event)
+        {:show-dialog :file-not-in-project}))
+
+    :resource
+    (when-some [selected-resources (not-empty (resource-dialog/make
+                                                workspace project
+                                                {:ext (:filter element)
+                                                 :selection :multiple}))]
+      (add-list-elements selected-resources map-event))
+
+    ;; default
+    (add-list-elements [(form/field-default element)] map-event)))
 
 (defmethod handle-event :remove-list-selection [{:keys [on-removed state-path ui-state]}]
   (let [indices-path (conj state-path :selected-indices)]
     [[:dispatch (assoc on-removed :fx/event (set (get-in ui-state indices-path)))]
-     [:set-ui-state (assoc-in ui-state indices-path [])]]))
+     [:set-ui-state (-> ui-state
+                        (assoc-in indices-path [])
+                        (update-in state-path dissoc :edit))]]))
```

Comment:
This fixes an issue where garbage would appear in list views when you removed an element with an active text selection.